### PR TITLE
Tests app/services/stripe_payment_status.rb

### LINKED ISSUE #11962: [Integration/API tests] app/services/stripe_payment_status.rb
Follow-up from https://github.com/openfoodfoundation/openfoodnetwork/pull/11900#issuecomment-1841892371.

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Related to Epic https://github.com/openfoodfoundation/openfoodnetwork/issues/11890.
Closes #11962

Tests `app/services/stripe_payment_status.rb` by replacing `receive_message_chain` Rspec mocks with responses from VCR calls.

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by filipefurtad0** [comment]
This PR seems to interfere with other VCR tapes, I'll move this back to In Dev till this is clarified.

File: spec/base_spec_helper.rb
Comment (filipefurtad0) [inline spec/base_spec_helper.rb:167]: It seems VCR was looking for a cassette in the Stripe folder, although this one is not related to Stripe:

```
       VCR is currently using the following cassette:
         - /home/runner/work/openfoodnetwork/openfoodnetwork/spec/fixtures/vcr_cassettes/Stripe-v10.3.0/ConnectAppJob/stores_connection_data_on_the_app.yml
```

Somehow, the around block on base spec helper is not really resetting the VCR configs. This change might just do it, but I'm not sure there isn't a better way.

Comment (dacook) [reply spec/base_spec_helper.rb:174]: This makes sense: we are setting a different config value, and need to re-set it back after the example is completed. This would be relevant for the other config change too (`default_cassette_options`).

I couldn't find a built-in method for achieving this, so would be tempted to build one..
For  now, I'd suggest saving the current vcr_config before `example.run`, then applying it again after.

I've pushed a new commit to demonstrate, let me know what you think.

File: spec/services/stripe_payment_status_spec.rb
Comment (dacook) [inline spec/services/stripe_payment_status_spec.rb:17]: I was going to suggest that we should reset the api key back  after the end of the test. But then I realised that this key is the only key that would be used during tests, so we might as well set it once in envrionment config. 

Then I realised we already do, in `STRIPE_INSTANCE_SECRET_KEY`. Do you know why we set the test key in a special `STRIPE_SECRET_TEST_API_KEY`?

**Review by dacook — APPROVED** [review]
Great, the spec updates look good. 

I've added a suggested improvement, and have a comment about further simplifying the setup.

I think this is good as-is, but will await a second review for my code updates.

**Review by mkllnk — APPROVED** [review]
Good team work.

**Pull request merged**
